src/main/kotlin/at/bitfire/dav4jvm/property/webdav/Depth.kt

```diff
@@ -0,0 +1,36 @@
+package at.bitfire.dav4jvm.property.webdav
+
+import at.bitfire.dav4jvm.Property
+import at.bitfire.dav4jvm.PropertyFactory
+import at.bitfire.dav4jvm.XmlReader
+import org.xmlpull.v1.XmlPullParser
+
+/**
+ * Represents a [NS_WEBDAV]`:depth` property.
+ */
+data class Depth(
+    /** May be `0`, `1` or [Int.MAX_VALUE] (infinite). */
+    val depth: Int? = null
+): Property {
+
+    companion object {
+
+        @JvmField
+        val NAME = Property.Name(NS_WEBDAV, "depth")
+
+    }
+
+
+    object Factory: PropertyFactory {
+
+        override fun getName() = NAME
+
+        override fun create(parser: XmlPullParser): Depth {
+            val text = XmlReader(parser).readText()
+            val level = if (text == "infinite") Int.MAX_VALUE else text?.toIntOrNull()
```

Comment:
When validating, I just found that [the literal should be `infinity`](http://www.webdav.org/specs/rfc4918.html#rfc.section.10.2) (only for `depth`; for `sync-level` it's `infinite`).

Will update that in the push draft (https://github.com/bitfireAT/webdav-push/issues/85).

---

Also for both elements case-insensitive comparison could be good.